reverse_engineering/databaseService/databaseService.js

```diff
@@ -332,105 +345,126 @@ const getTableForeignKeys = async (connectionClient, dbName, logger) => {
 			skip: true,
 		},
 		logger,
+	});
+
+	const schemaAlias = 'sch';
+	const table1Alias = 'tab1';
+
+	const whereClauseParts = flatMap(
+		Object.entries(tablesInfo).map(([schemaName, tableNames]) =>
+			tableNames.map(
+				tableName => `(${table1Alias}.name = '${tableName}' AND ${schemaAlias}.name = '${schemaName}')`,
+			),
+		),
 	);
+
 	return mapResponse(
-		await currentDbConnectionClient.query`
-		SELECT obj.name AS FK_NAME,
-				sch.name AS [schema_name],
-				tab1.name AS [table],
+		await currentDbConnectionClient.query(`
+			SELECT obj.name AS FK_NAME,
+				${schemaAlias}.name AS [schema_name],
+				${table1Alias}.name AS [table],
 				col1.name AS [column],
 				tab2.name AS [referenced_table],
 				col2.name AS [referenced_column],
 				fk.delete_referential_action_desc AS on_delete,
 				fk.update_referential_action_desc AS on_update
-		FROM sys.foreign_key_columns fkc
-		INNER JOIN sys.objects obj
-			ON obj.object_id = fkc.constraint_object_id
-		INNER JOIN sys.tables tab1
-			ON tab1.object_id = fkc.parent_object_id
-		INNER JOIN sys.schemas sch
-			ON tab1.schema_id = sch.schema_id
-		INNER JOIN sys.columns col1
-			ON col1.column_id = parent_column_id AND col1.object_id = tab1.object_id
-		INNER JOIN sys.tables tab2
-			ON tab2.object_id = fkc.referenced_object_id
-		INNER JOIN sys.columns col2
-			ON col2.column_id = referenced_column_id AND col2.object_id = tab2.object_id
-		INNER JOIN sys.foreign_keys fk
-			ON fk.object_id = obj.object_id
-		`,
+			FROM sys.foreign_key_columns fkc
+			INNER JOIN sys.objects obj
+				ON obj.object_id = fkc.constraint_object_id
+			INNER JOIN sys.tables ${table1Alias}
+				ON ${table1Alias}.object_id = fkc.parent_object_id
+			INNER JOIN sys.schemas ${schemaAlias}
+				ON ${table1Alias}.schema_id = ${schemaAlias}.schema_id
+			INNER JOIN sys.columns col1
+				ON col1.column_id = parent_column_id AND col1.object_id = ${table1Alias}.object_id
+			INNER JOIN sys.tables tab2
+				ON tab2.object_id = fkc.referenced_object_id
+			INNER JOIN sys.columns col2
+				ON col2.column_id = referenced_column_id AND col2.object_id = tab2.object_id
+			INNER JOIN sys.foreign_keys fk
+				ON fk.object_id = obj.object_id
+			WHERE ${whereClauseParts.join(' OR ')}
```

Comment:
So you have 2 options for restricting queries:

- using a `WHERE` clause, like you did here
- using a `WITH` clause, like we did in Synapse

I think that the `WITH` clause is evaluated BEFORE the joins, whereas your `WHERE` clause is evaluated after the joins. Could you please compare the execution plan for each option and report the respective costs here? We need to choose the most performant option...